spec/docs/termmap.md

```diff
@@ -162,15 +111,57 @@ The value MUST be an [IRI]() and MUST be one of the following options:
 * If the term map is an [=object map=]: `rml:IRI`, `rml:BlankNode`, or `rml:Literal`
 * If the term map is a [=graph map=]: `rml:IRI`
 
+### Default Term Types
 
 If the [=term map=] does not have a `rml:termType` property, then its [=term type=] is:
 
+* `rml:IRI`, if it is a [=subject map=], [=predicate map=] or [=graph map=]
 * `rml:Literal`, if it is an [=object map=]
 and at least one of the following conditions is true:
 * It is a [=reference-valued term map=].
     * It has a `rml:languageMap` property (and thus a [specified language tag]()).
     * It has a `rml:datatypeMap` property (and thus a [specified datatype]()).
-* rml:IRI, otherwise.
+* `rml:IRI`, otherwise.
+
+### Explicitly Defined Term Types
+
+To change the default [=term type=] of a [=subject map=] or [=graph map=]
+to a blank node, the [=term type=] MUST be explicitly defined to be a `rml:BlankNode`.
+
+To change the default [=term type=] of an [=object map=], the [=term type=] MUST be explicitly defined:
+* If the [=term type=] is `rml:IRI`, an IRI will be generated;
+* If the [=term type=] is `rml:BlankNode`, a Blank Node will be generated.
+
+If the [=term type=] is explicitly defined to be a `rml:BlankNode`,
+a [=term map=] MAY not have an [=expression map=].
+Then an RML Processor MUST generate a random value for the blank node.   
+
+### IRI encoding
+
+If the [=term type=] of the [=template-valued term map=] is `rml:IRI`, then a [=reference value transforming function=] should be applied during the evaluation of the [=template expression=]. The [=reference value transforming function=] should transform a [=reference value=] into an [=IRI-safe version=] of that value.
+
+The <dfn data-lt="IRI-safe">IRI-safe version</dfn> of a string is obtained by applying the following transformation
+to any character that is not in the iunreserved production in [[RFC3987]]():
+
+1. Convert the character to a sequence of one or more octets using UTF-8 [[RFC3629]]()
+2. Percent-encode each octet [[RFC3986]]()
+
+The following table shows examples of strings and their IRI-safe versions:
+
+
+| String | IRI-safe version  |
+| :-------------| :-----|
+| 42 | 42 | 
+| Hello World! | Hello%20World%21 |
+| 2011-08-23T22:17:00Z | 2011-08-23T22%3A17%3A00Z |
+| ~A_17.1-2 |  ~A_17.1-2 |
+
+<aside class="note">
+RML always performs percent-encoding when IRIs are generated from [=string templates=].
+If IRIs need to be generated without percent-encoding,
+then `rml:reference` should be used instead of `rml:template`,
+with a [=logical source=] that performs the string concatenation.
+</aside>
```

Comment:
Why has this been moved outside of the section about template-valued term maps? This description is specifically about template-valued term maps. I think this makes the spec less easy to read if we split this off.